mcs/class/Mono.Profiler.Log/Test/ProfilerTestRun.cs

```diff
@@ -37,16 +37,28 @@ public override void VisitBefore (LogEvent ev)
 
 		static volatile int _id;
 
+		static string _tempDirectory;
 		static string _testAssemblyPath;
 		static Process _currentProcess;
 
 		public ProfilerTestRun (string name, string options)
 		{
-			_testAssemblyPath = Path.Combine (Path.GetDirectoryName (System.Reflection.Assembly.GetExecutingAssembly ().Location), "log-profiler-test.exe");
-			_currentProcess = Process.GetCurrentProcess();
 			Name = name;
 			Options = options;
-			_output = $"test-{_id++}.mlpd";
+			_output = Path.Combine (_tempDirectory, $"test-{_id++}.mlpd");
+		}
+
+		static ProfilerTestRun ()
+		{
+			_testAssemblyPath = Path.Combine (Path.GetDirectoryName (System.Reflection.Assembly.GetExecutingAssembly ().Location), "log-profiler-test.exe");
+			_currentProcess = Process.GetCurrentProcess();
+			_tempDirectory = Path.Combine (Path.GetTempPath(), Path.GetRandomFileName ());
+			Directory.CreateDirectory (_tempDirectory);
+		}
+
+		~ProfilerTestRun()
+		{
+			File.Delete (_output);
```

Comment:
Given that profiler test failures are often hard to reproduce, I prefer to keep the MLPDs around when possible to help diagnose a failure. This is mostly useful locally, so maybe we could delete the files only on CI, if doing so is actually necessary?

Reply:
For the same reasons, I'm not sure it's a good idea to just throw these files in a random/unknowable directory in `/tmp`.

Reply:
Hmm, now that I think about it more I'm not sure I really need the files to be outside the working directory. This was part of my work on getting the tests running in the Helix environment but I probably only need to make `_output` a full instead of relative path. I'll try and come back :)

Reply:
Yup confirmed, using Path.GetFullPath() is enough. Updated.